iOS: Updating matrix file fails with java.lang.NumberFormatException: For input string: "167,793"

```
java -jar ./flank.jar -v
v4.2-SNAPSHOT
a7d6d719225208f5d4c0cac72ab23ea8a31ea3f1
```

There is an exception happens in the end of FetchArtifacts when it starts updating matrix file. So it does not go further and does not upload smartFlank .xml file. 

```
FetchArtifacts
  .......................................................
  Updating matrix file

Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.ios.IosRunCommand@2dbc408c): java.lang.NumberFormatException: For input string: "167,793"
	at picocli.CommandLine.execute(CommandLine.java:1168)
	at picocli.CommandLine.access$800(CommandLine.java:141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at picocli.CommandLine.run(CommandLine.java:1974)
	at picocli.CommandLine.run(CommandLine.java:1920)
	at ftl.Main$Companion.main(Main.kt:48)
	at ftl.Main.main(Main.kt)
Caused by: java.lang.NumberFormatException: For input string: "167,793"
	at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
	at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
	at java.base/java.lang.Double.parseDouble(Double.java:543)
	at ftl.reports.xml.model.JUnitTestSuite.mergeDouble(JUnitTestSuite.kt:54)
	at ftl.reports.xml.model.JUnitTestSuite.merge(JUnitTestSuite.kt:65)
	at ftl.reports.xml.model.JUnitTestResult.merge(JUnitTestResult.kt:40)
	at ftl.reports.util.ReportManager.processXml(ReportManager.kt:62)
	at ftl.reports.util.ReportManager.parseTestSuite(ReportManager.kt:71)
	at ftl.reports.util.ReportManager.generate(ReportManager.kt:79)
	at ftl.run.TestRunner.newRun(TestRunner.kt:356)
	at ftl.run.TestRunner$newRun$1.invokeSuspend(TestRunner.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.ResumeModeKt.resumeUninterceptedMode(ResumeMode.kt:45)
	at kotlinx.coroutines.internal.ScopeCoroutine.onCompletionInternal$kotlinx_coroutines_core(Scopes.kt:28)
	at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:305)
	at kotlinx.coroutines.JobSupport.tryFinalizeFinishingState(JobSupport.kt:230)
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:799)
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:742)
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:117)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:45)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.kt:116)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:76)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:53)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at ftl.cli.firebase.test.ios.IosRunCommand.run(IosRunCommand.kt:31)
	at picocli.CommandLine.execute(CommandLine.java:1160)
	... 9 more
```

--- a/test_runner/src/main/kotlin/ftl/reports/xml/model/JUnitTestSuite.kt
+++ b/test_runner/src/main/kotlin/ftl/reports/xml/model/JUnitTestSuite.kt
@@ -46,12 +46,22 @@ data class JUnitTestSuite(
     val systemErr: Any? // <system-err />
 ) {
 
+    /**
+     * Strips all characters except numbers and a period
+     * Returns 0 when the string is null
+     *
+     * Example: z1,23.45 => 123.45 */
+    private fun String?.clean(): String {
+        if (this == null) return "0"
+        return this.replace(Regex("""[^0-9\\.]"""), "")
+    }
+
     private fun mergeInt(a: String?, b: String?): String {
-        return ((a ?: "0").toInt() + (b ?: "0").toInt()).toString()
+        return (a.clean().toInt() + b.clean().toInt()).toString()
     }
 
     private fun mergeDouble(a: String?, b: String?): String {
-        return "%.3f".format(((a ?: "0").toDouble() + (b ?: "0").toDouble()))
+        return "%.3f".format((a.clean().toDouble() + b.clean().toDouble()))
     }
 
     fun merge(other: JUnitTestSuite): JUnitTestSuite {

--- a/test_runner/src/test/kotlin/ftl/fixtures/ftl_junit_xml/ios_large_num.xml
+++ b/test_runner/src/test/kotlin/ftl/fixtures/ftl_junit_xml/ios_large_num.xml
@@ -1,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<testsuites>
+  <testsuite name='EarlGreyExampleSwiftTests' hostname='localhost' tests='a99,999' failures='z99,999' errors='t99,999' time='o99,999.881'>
+    <properties />
+    <testcase name='testBasicSelectionActionAssert()' classname='EarlGreyExampleSwiftTests' time='0' />
+    <system-out />
+    <system-err />
+  </testsuite>
+</testsuites>

--- a/test_runner/src/test/kotlin/ftl/reports/xml/JUnitXmlTest.kt
+++ b/test_runner/src/test/kotlin/ftl/reports/xml/JUnitXmlTest.kt
@@ -12,6 +12,7 @@ class JUnitXmlTest {
         val androidFailXml = Paths.get("$xmlRoot/android_fail.xml")!!
         val iosPassXml = Paths.get("$xmlRoot/ios_pass.xml")!!
         val iosFailXml = Paths.get("$xmlRoot/ios_fail.xml")!!
+        val iosLargeNum = Paths.get("$xmlRoot/ios_large_num.xml")!!
         val androidSkipped = """
 <testsuite name="" tests="2" failures="0" errors="0" skipped="1" time="0.026" timestamp="2018-10-26T19:57:28" hostname="localhost">
     <properties/>
@@ -90,6 +91,23 @@ junit.framework.Assert.fail(Assert.java:50)</failure>
     }
 
     @Test
+    fun `Merge iOS large time`() {
+        val merged = parseAllSuitesXml(iosLargeNum).merge(parseAllSuitesXml(iosLargeNum)).xmlToString()
+
+        val expected = """
+<?xml version='1.0' encoding='UTF-8' ?>
+<testsuites>
+  <testsuite name="EarlGreyExampleSwiftTests" tests="199998" failures="199998" errors="199998" skipped="0" time="199999.762" hostname="localhost">
+    <testcase name="testBasicSelectionActionAssert()" classname="EarlGreyExampleSwiftTests" time="0"/>
+    <testcase name="testBasicSelectionActionAssert()" classname="EarlGreyExampleSwiftTests" time="0"/>
+  </testsuite>
+</testsuites>
+
+        """.trimIndent()
+        assertThat(merged).isEqualTo(expected)
+    }
+
+    @Test
     fun parse_androidSkipped() {
         val parsed = parseOneSuiteXml(androidSkipped)
         assertThat(parsed.testsuites?.first()?.testcases?.first()?.skipped).isNull()
@@ -101,7 +119,8 @@ junit.framework.Assert.fail(Assert.java:50)</failure>
         merged.merge(merged)
         val actual = merged.xmlToString()
 
-        assertThat(actual).isEqualTo("""
+        assertThat(actual).isEqualTo(
+            """
 <?xml version='1.0' encoding='UTF-8' ?>
 <testsuites>
   <testsuite name="" tests="4" failures="0" errors="0" skipped="2" time="0.052" timestamp="2018-10-26T19:57:28" hostname="localhost">
@@ -116,7 +135,8 @@ junit.framework.Assert.fail(Assert.java:50)</failure>
   </testsuite>
 </testsuites>
 
-        """.trimIndent())
+        """.trimIndent()
+        )
     }
 
     @Test

